agent/engine/engine_integ_test.go

```diff
@@ -937,3 +937,64 @@ func TestSweepContainer(t *testing.T) {
 		t.Error("Expected container to have been sweept but was not")
 	}
 }
+
+// This integ test is meant to validate the docker assumptions related to
+// https://github.com/aws/amazon-ecs-agent/issues/261
+// Namely, this test verifies that Docker does emit a 'die' event after an OOM
+// event if the init dies.  It also verifies that we don't see OOM as fatal
+// unless ther's also a die event.
+func TestOOMHasDieEvent(t *testing.T) {
+	// Verify that docker does emit a 'die' event after an 'oom' event
+	taskEngine, done := setup(t)
+	defer done()
+
+	taskEvents, contEvents := taskEngine.TaskEvents()
+	defer discardEvents(taskEvents)()
+
+	testTask := createTestTask("oomtest")
+	testTask.Containers[0].Image = "127.0.0.1:51670/busybox"
+	testTask.Containers[0].Memory = 20
+	testTask.Containers[0].EntryPoint = &[]string{"sh", "-c"}
+	testTask.Containers[0].Command = []string{`
+	dd if=/dev/zero of=/dev/null bs=25M count=1
+	sleep 1
+	exec dd if=/dev/zero of=/dev/null bs=25M count=1`}
+	// Regular dd spawns a subprocess we expect to get OOM killed, but not take down the whole container
+	// exec'd dd becomes pid1 so it should take down the container
+
+	start := time.Now()
+
+	go taskEngine.AddTask(testTask)
+
+	expected_events := []api.ContainerStatus{api.ContainerRunning, api.ContainerStopped}
+
+	var contEvent api.ContainerStateChange
+	for contEvent = range contEvents {
+		if contEvent.TaskArn != testTask.Arn {
+			continue
+		}
+
+		expected_event := expected_events[0]
+		expected_events = expected_events[1:]
+		if contEvent.Status != expected_event {
+			t.Error("Got event " + contEvent.Status.String() + " but expected " + expected_event.String())
+		}
+		if len(expected_events) == 0 {
+			break
+		}
+	}
+
+	if contEvent.ExitCode == nil {
+		t.Error("Expected exitcode to be set")
+	} else if *contEvent.ExitCode != 137 {
+		t.Errorf("Expected exitcode to be 137, not %v", *contEvent.ExitCode)
+	}
+
+	if !strings.HasPrefix(contEvent.Reason, OutOfMemoryError{}.ErrorName()) {
+		t.Errorf("Expected reason to have OOM error, was: %v", contEvent.Reason)
+	}
+
+	if time.Since(start) < time.Second {
```

Comment:
I don't think this is strong-enough verification that the first `dd` didn't cause us to emit `api.ContainerStopped`.